agents.md

```diff
@@ -0,0 +1,319 @@
+# Agents Guide
+
+This guide helps AI coding agents and automation tools work effectively with the craft-application repository.
+
+## Overview
+
+**craft-application** is a Python library that serves as the base framework for all Starcraft applications (like snapcraft, charmcraft, etc.). It provides common mechanisms for application services, project models, and other shared functionality.
+
+- **Language**: Python 3.10+
+- **License**: LGPL-3.0
+- **Build System**: setuptools with setuptools_scm
+- **Package Manager**: uv
+- **Documentation**: See [README.md](README.md) and [CONTRIBUTING.md](CONTRIBUTING.md)
+
+## Development Workflow
+
+### Initial Setup
+
+1. **Clone the repository** (see [CONTRIBUTING.md](CONTRIBUTING.md) for details):
+
+    ```bash
+    git clone https://github.com/canonical/craft-application --recurse-submodules
+    cd craft-application
+    ```
+
+2. **Set up the development environment**:
+
+    ```bash
+    make setup
+    ```
+
+    This installs all dependencies, sets up the virtual environment, and configures pre-commit hooks.
+
+3. **Verify the setup**:
+    ```bash
+    make lint
+    make test
+    ```
+
+### Common Development Tasks
+
+Use these `make` targets for everyday development. Run `make help` to see all available targets.
+
+#### Formatting Code
+
+```bash
+make format                # Run all automatic formatters
+```
+
+#### Linting Code
+
+```bash
+make lint                  # Run all linters
+```
+
+#### Running Tests
+
+```bash
+make test                  # Run all tests
+```
+
+To run tests directly within the uv-created virtual environment (useful when only modifying tests):
+
+```bash
+# Activate the virtual environment (if not already activated)
+source .venv/bin/activate
+
+# Run tests in a specific file
+pytest tests/unit/test_application.py
+
+# Run tests matching a pattern
+pytest -k "test_pattern"
+
+# Run with verbose output
+pytest -v
+
+# Run in parallel (if pytest-xdist is available)
+pytest -n auto
+```
+
+#### Building Documentation
+
+```bash
+make docs                  # Build documentation
+```
+
+#### Cleaning Up
+
+```bash
+make clean                 # Remove build artifacts and temporary files
+```
+
+### Pre-commit Hooks
+
+This project uses pre-commit hooks that run automatically on `git commit`. They include:
+
+- Ruff linting and formatting
+- Prettier formatting for YAML/JSON/Markdown
+- Trailing whitespace removal
+- End-of-file fixer
+- Various file checks
+
+See [.pre-commit-config.yaml](.pre-commit-config.yaml) for the full configuration.
```

Comment:
This is unnecessary except to ensure that the agent installs the pre-commit hooks and uses them before committing.